packages/typespec-azure-resource-manager/test/rules/arm-common-types-version.test.ts

```diff
@@ -20,42 +20,6 @@ describe("typespec-azure-resource-manager: ARM common-types version rule", () =>
     );
   });
 
-  it("emits diagnostic when a version in the enum is missing a common type version", async () => {
```

Comment:
what does this do now?

Reply:
This is rule is now obsolete as we will add default versions.

Reply:
So shouldn't the whole rule be removed? And isn't the point of this rule is we didn't want to use default?

Reply:
> And isn't the point of this rule is we didn't want to use default?
My implementation has geared towards no breaking changes. But IF this is indeed a strong goal we have, I can adjust the logic to call @useDependency directly for missing declarations.

Reply:
You feedback is taken, I have put the rule back. It checks the existence of decorator rather than stateMap values.